### PR TITLE
Dịch "linear-regression-scratch_vn" phần 8

### LINKED ISSUE #624: Dịch "linear-regression-scratch_vn" - Phần 8
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/linear-regression-scratch_vn.md#L348-L402

### PR DESCRIPTION
**Pull request by patrickphatnguyen**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #624 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm [hướng dẫn hoặc yêu cầu dịch hình](../img/README.md).

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](../docs/contributors_info.md/)**.

## Với reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 1]
```diff
@@ -360,24 +360,31 @@ In summary, we will execute the following loop:
...
+    * Cập nhật bộ tham số $(\mathbf{w}, b) \leftarrow (\mathbf{w}, b) - \eta \mathbf{g}$
+
 
 <!--
 In the code below, `l` is a vector of the losses for each example in the minibatch.
 Because `l` is not a scalar variable, running `l.backward()` adds together the elements in `l` to obtain the new variable and then calculates the gradient.
 -->
 
-*dịch đoạn phía trên*
+Trong đoạn code dưới đây, `l` là một vector độ mất mát với từng các mẫu trong minibatch.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Trong đoạn code dưới đây, `l` là một vector độ mất mát của từng mẫu trong minibatch.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 2]
```diff
@@ -360,24 +360,31 @@ In summary, we will execute the following loop:
...
+Vì `l` không phải là biến vô hướng, chạy `l.backward()`để cộng các phần tử trong `l` để tạo ra một biến mới và sau đó mới tính gradient.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Vì `l` không phải là biến vô hướng, chạy `l.backward()` sẽ cộng các phần tử trong `l` để tạo ra một biến mới và sau đó mới tính gradient.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 3]
```diff
@@ -360,24 +360,31 @@ In summary, we will execute the following loop:
...
 
 <!--
 In each epoch (a pass through the data), we will iterate through the entire dataset (using the `data_iter` function) 
 once passing through every examples in the training dataset (assuming the number of examples is divisible by the batch size).
 The number of epochs `num_epochs` and the learning rate `lr` are both hyper-parameters, which we set here to $3$ and $0.03$, respectively.
 Unfortunately, setting hyper-parameters is tricky and requires some adjustment by trial and error.
 We elide these details for now but revise them later in :numref:`chap_optimization`.
--->
+-->
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
-->
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 4]
```diff
@@ -360,24 +360,31 @@ In summary, we will execute the following loop:
...
 In each epoch (a pass through the data), we will iterate through the entire dataset (using the `data_iter` function) 
 once passing through every examples in the training dataset (assuming the number of examples is divisible by the batch size).
 The number of epochs `num_epochs` and the learning rate `lr` are both hyper-parameters, which we set here to $3$ and $0.03$, respectively.
 Unfortunately, setting hyper-parameters is tricky and requires some adjustment by trial and error.
 We elide these details for now but revise them later in :numref:`chap_optimization`.
--->
+--> 
 
-*dịch đoạn phía trên*
+Với mỗi epoch (một lần chạy qua tập dữ liệu), chúng ta sẽ lặp qua toàn bộ tập dữ liệu (sử dụng hàm `data_iter`) cho đến khi đi qua toàn bộ mọi mẫu trong tập huấn luyện (giả định rằng số mẫu chia hết cho kích thước batch). Số epoch `num_epochs` và tốc độ học `lr` đều là siêu tham số, mà chúng ta đặt ở đây là tương ứng $3$ and $0.03$.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Với mỗi epoch (một lần chạy qua tập dữ liệu), chúng ta sẽ lặp qua toàn bộ tập dữ liệu (sử dụng hàm `data_iter`) cho đến khi đi qua toàn bộ mọi mẫu trong tập huấn luyện (giả định rằng số mẫu chia hết cho kích thước batch).
Số epoch `num_epochs` và tốc độ học `lr` đều là siêu tham số, mà chúng ta đặt ở đây là tương ứng $3$ and $0.03$.
```
Mỗi câu một dòng nhé

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 5]
```diff
@@ -360,24 +360,31 @@ In summary, we will execute the following loop:
...
+Không may thay, việc lựa chọn siêu tham số khó và đòi hỏi một vài sự thay đổi bằng việc thử và sai.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Không may thay, việc lựa chọn siêu tham số là khó và đòi hỏi một vài sự điều chỉnh bằng cách thử và sai.
```

**Comment by patrickphatnguyen** [comment]
Các góp ý của em đã được resolved nha. Anh sẽ double check lại bản dịch, thanks em @rootonchair

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 6]
```diff
@@ -360,14 +360,19 @@ In summary, we will execute the following loop:
     * Update parameters $(\mathbf{w}, b) \leftarrow (\mathbf{w}, b) - \eta \mathbf{g}$
 -->
 
-*dịch đoạn phía trên*
+* Khởi tạo bộ tham số $(\mathbf{w}, b)$
+* Lặp lại cho tới khi hoàn thành
+    * Tính gradient $\mathbf{g} \leftarrow \partial_{(\mathbf{w},b)} \frac{1}{\mathcal{B}} \sum_{i \in \mathcal{B}} l(\mathbf{x}^i, y^i, \mathbf{w}, b)$
+    * Cập nhật bộ tham số $(\mathbf{w}, b) \leftarrow (\mathbf{w}, b) - \eta \mathbf{g}$
+
```
Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 7]
```diff
@@ -360,14 +360,19 @@ In summary, we will execute the following loop:
...
 
 <!--
 In the code below, `l` is a vector of the losses for each example in the minibatch.
 Because `l` is not a scalar variable, running `l.backward()` adds together the elements in `l` to obtain the new variable and then calculates the gradient.
 -->
 
-*dịch đoạn phía trên*
+Trong đoạn code dưới đây, `l` là một vector độ mất mát của từng mẫu trong minibatch.
```
Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Trong đoạn mã dưới đây, `l` là một vector của các mất mát của từng mẫu trong minibatch.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 8]
```diff
@@ -377,7 +382,10 @@ Unfortunately, setting hyper-parameters is tricky and requires some adjustment b
 We elide these details for now but revise them later in :numref:`chap_optimization`.
 -->
 
-*dịch đoạn phía trên*
+Với mỗi epoch (một lần chạy qua tập dữ liệu), chúng ta sẽ lặp qua toàn bộ tập dữ liệu (sử dụng hàm `data_iter`) cho đến khi đi qua toàn bộ mọi mẫu trong tập huấn luyện (giả định rằng số mẫu chia hết cho kích thước batch).
```
Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]: @lkhphuc mình có dịch từ `epoch` không nhỉ?

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 9]
```diff
@@ -377,7 +382,10 @@ Unfortunately, setting hyper-parameters is tricky and requires some adjustment b
...
+Số epoch `num_epochs` và tốc độ học `lr` đều là siêu tham số, mà chúng ta đặt ở đây là tương ứng $3$ and $0.03$.
```
Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Số epoch `num_epochs` và tốc độ học `lr` đều là siêu tham số, mà chúng ta đặt ở đây là tương ứng $3$ và $0.03$.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 10]
```diff
@@ -377,7 +382,10 @@ Unfortunately, setting hyper-parameters is tricky and requires some adjustment b
...
+Không may thay, việc lựa chọn siêu tham số là khó và đòi hỏi một vài sự điều chỉnh bằng cách thử và sai.
```
Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Không may thay, việc lựa chọn siêu tham số thường không đơn giản và đòi hỏi một vài sự điều chỉnh bằng cách thử và sai.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 8 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Theo em nhớ là không anh ạ.
